packages/extension-base/src/koni/background/handlers/State.ts

```diff
@@ -1177,6 +1182,345 @@ export default class KoniState {
+      recipient: transaction.to as string,
+      sender: account.address,
+      utxos
+    };
+
+    if (!transaction.to) {
+      throw new Error();
+    }
+
+    const fallbackCalculate = (recipients: string[]) => {
+      const utxos = filterUneconomicalUtxos({
+        utxos: determineUtxosArgs.utxos,
+        feeRate: determineUtxosArgs.feeRate,
+        recipients,
+        sender: determineUtxosArgs.sender
+      });
+
+      const { txVBytes: vSize } = getSizeInfo({
+        inputLength: utxos.length || 1,
+        sender: fromAddress,
+        recipients
+      });
+
+      return {
+        vSize,
+        maxTransferable: utxos.reduce((previous, input) => previous.plus(input.value), new BigN(0)),
+        estimatedFee: Math.ceil(determineUtxosArgs.feeRate * vSize).toString()
+      };
+    };
+
+    const getBalance = async (senderAddress: string) => {
+      const filteredUtxos = await getTransferableBitcoinUtxos(bitcoinApi, senderAddress);
+
+      let balanceValue = new BigN(0);
+
+      filteredUtxos.forEach((utxo) => {
+        balanceValue = balanceValue.plus(utxo.value);
+      });
+
+      return balanceValue;
+    };
+
+    let maxTransferable = new BigN('0');
+    let estimatedFee = '0';
+
+    try {
+      const { fee: _estimatedFee, inputs } = determineUtxosForSpend(determineUtxosArgs);
+
+      const { txVBytes: vSize } = getSizeInfo({
+        inputLength: inputs.length,
+        sender: fromAddress,
+        recipients: [transaction.to]
+      });
+
+      estimatedFee = new BigN(_estimatedFee).toFixed(0);
+      feeOptions = {
+        ...feeOptions_,
+        estimatedFee,
+        vSize
+      };
+    } catch (_e) {
+      if (!feeOptions) {
+        const fb = fallbackCalculate([transaction.to, transaction.to]);
+
+        estimatedFee = fb.estimatedFee;
+
+        feeOptions = {
+          ...feeOptions_,
+          estimatedFee,
+          vSize: fb.vSize
+        };
+      }
+    }
+
+    maxTransferable = await getBalance(fromAddress);
+
+    // Validate balance
+    if (maxTransferable.lt(new BigN(estimatedFee).plus(new BigN(autoFormatNumber(transactionParams.recipients[0].amount) || '0')))) {
+      throw new EvmProviderError(EvmProviderErrorType.INVALID_PARAMS, t('Insufficient balance'));
+    }
+
+    const requestPayload: BitcoinSendTransactionRequest = {
+      ...transaction,
+      hashPayload: JSON.stringify(transaction),
+      fee: feeOptions,
+      account: account,
+      canSign: true
+    };
+
+    const eType = transaction.value ? ExtrinsicType.TRANSFER_BALANCE : ExtrinsicType.EVM_EXECUTE;
+
+    const transactionData = { ...transaction };
+    const token = this.chainService.getNativeTokenInfo(networkKey);
+
+    if (eType === ExtrinsicType.TRANSFER_BALANCE) {
+      // @ts-ignore
+      transactionData.tokenSlug = token.slug;
+    }
+
+    // Custom handle this instead of general handler transaction
+    const transactionEmitter = await this.transactionService.addTransaction({
```

Comment:
Should split to another handler, shouldn't reuse logic send transaction in extension